model/src/main/kotlin/org/digma/intellij/plugin/ui/model/insights/InsightGroupListViewItem.kt

```diff
@@ -2,6 +2,6 @@ package org.digma.intellij.plugin.ui.model.insights
 
 import org.digma.intellij.plugin.ui.model.listview.GroupListViewItem
 
-open class InsightGroupListViewItem(groupId: String, val type: InsightGroupType) :
-    GroupListViewItem(type.sortIndex, groupId) {
+open class InsightGroupListViewItem(groupId: String, val type: InsightGroupType, route: String) :
+    GroupListViewItem(type.sortIndex, groupId, route) {
```

Comment:
similar here, why does groupId is not enough?

Reply:
we need route for working of logic of correct grouping by the fullRouteName and providing relevant GroupViewModel:
`private fun createEndpointGroupViewModel(fullRouteName: String): GroupViewModel{
        val routeInfo = EndpointSchema.getRouteInfo(fullRouteName)
        val endpoint =  routeInfo.shortName
        if(routeInfo.schema == HTTP_SCHEMA){
            val split =endpoint.split(' ')
            return GroupViewModel(asHtml("${spanBold("HTTP")} ${span("${split[0].uppercase()} ${split[1]}")}"),  Laf.Icons.Insight.INTERFACE)
        }
        if(routeInfo.schema == RPC_SCHEMA){
            return GroupViewModel(asHtml(span(endpoint)),  Laf.Icons.Insight.INTERFACE)
        }
        if(routeInfo.schema == CONSUMER_SCHEMA){
            return GroupViewModel(asHtml(span(endpoint)),  Laf.Icons.Insight.MESSAGE)
        }
        return GroupViewModel(asHtml(""),  Laf.Icons.Insight.INTERFACE)
    }`